addons/account/models/chart_template.py

```diff
@@ -776,3 +755,27 @@ def _parse_csv(self, template_code, model, module=None):
             except FileNotFoundError:
                 _logger.debug("No file %s found for template '%s'", model, module)
         return res
+
+    def _load_translations(self, langs=False, companies=False):
+        langs = langs or [code for code, _name in self.env['res.lang'].get_installed()]
+        companies = companies or self.env['res.company'].search([('chart_template', '!=', False)])
+
+        chart2company = {}
+        for company in companies:
+            chart2company.setdefault(company.chart_template, self.env['res.company'])
+            chart2company[company.chart_template] += company
+
+        translation_importer = TranslationImporter(self.env.cr, verbose=False)
+        for chart_template, companies in chart2company.items():
```

Comment:
```suggestion
        translation_importer = TranslationImporter(self.env.cr, verbose=False)
        for chart_template, companies in groupby(companies, key=lambda c: c.chart_template):
```

since you don't need `companies` in the loop to be a recordset, you can use the `tools.groupby` to simplify the code
Maybe also rename one of the `companies`s to improve the readability